### PR TITLE
feat: Added ability to configure PR columns with parameters using filters.

### PR DESCRIPTION
**Pull request by marleypowell**
Created some initial filters that can be used to create pull request columns:
- All
- AssignedToUser
- AssignedToCurrentUser
- AssignedToTeam
- AssignedToTeams
- AssignedToCurrentUsersTeams
- CreatedByUser
- CreatedByCurrentUser
- CreatedByTeam
- CreatedByTeams
- CreatedByCurrentUsersTeams

Example custom column creation:
```tsx
const COLUMN_CONFIGS: PullRequestColumnConfig[] = [
  {
    title: 'Created by me',
    filters: [{ type: FilterType.CreatedByCurrentUser }],
  },
  {
    title: 'Created by Backstage Core',
    filters: [
      {
        type: FilterType.CreatedByTeam,
        teamName: 'Backstage Core',
      },
    ],
  },
  {
    title: 'Assigned to my teams',
    filters: [{ type: FilterType.AssignedToCurrentUsersTeams }],
  },
  {
    title: 'Other PRs',
    filters: [{ type: FilterType.All }],
    simplified: true,
  },
];

<Route
  path="/azure-pull-requests"
  element={<AzurePullRequestsPage projectName="{PROJECT_NAME}" defaultColumnConfigs={COLUMN_CONFIGS} />}
/>
```

Task from #7678

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by marleypowell — COMMENTED** [review]
Ready for review 🙂

File: plugins/azure-devops/src/components/PullRequestsPage/lib/filters/createdByUserFilter.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  const email = filter.email;
+
+  return (pullRequest: DashboardPullRequest): boolean => {
+    const uniqueName = pullRequest.createdBy?.uniqueName;
+
+    if (!email || !uniqueName) {
+      return false;
+    }
+
+    return equalsIgnoreCase(email, uniqueName);
```
[the remark this answers is not in the document]
Comment (awanlin) [reply plugins/azure-devops/src/components/PullRequestsPage/lib/filters/createdByUserFilter.ts:46]: OK, looks like we could use `identityApi.getProfile().displayName` (`getProfile` is depreciated so we should use `getProfileInfo`) on the Backstage side, then `pullRequest.createdBy?.displayName` for the Azure DevOps side, that works. Maybe we sue this only if their host isn't `dev.azure.com`?

**Review by benjdlambert — APPROVED** [review]

File: plugins/azure-devops/src/components/PullRequestsPage/lib/filters/createdByUserFilter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (marleypowell) [reply plugins/azure-devops/src/components/PullRequestsPage/lib/filters/createdByUserFilter.ts:46]: Oh ok that's annoying. I will have a further look into this and iterate on the approach in a follow up PR.

**Comment by benjdlambert** [comment]
Thanks all! Nice work! The contributions on these plugins is great!

**Pull request merged**